src/crypto/slow-hash.c

```diff
@@ -61,16 +61,33 @@ extern int aesb_pseudo_round(const uint8_t *in, uint8_t *out, const uint8_t *exp
 #define VARIANT1_2(p) \
   do if (variant > 0) \
   { \
-    ((uint32_t*)p)[2] ^= nonce; \
+    xor64(p, tweak1_2); \
   } while(0)
 
-#define VARIANT1_INIT() \
-  if (variant > 0 && length < 43) \
+#define VARIANT1_CHECK() \
+  do if (length < 43) \
   { \
     fprintf(stderr, "Cryptonight variants need at least 43 bytes of data"); \
     _exit(1); \
+  } while(0)
+
+#define NONCE_POINTER (((const uint8_t*)data)+35)
+
+#define VARIANT1_PORTABLE_INIT() \
+  uint8_t tweak1_2[8]; \
+  do if (variant > 0) \
+  { \
+    VARIANT1_CHECK(); \
+    memcpy(&tweak1_2, &state.hs.b[192], sizeof(tweak1_2)); \
+    xor64(tweak1_2, NONCE_POINTER); \
```

Comment:
Why do you XOR with the address of the pointer? Should it be `*((const uint64_t*)NONCE_POINTER))`

Reply:
@psychocrypt VARIANT1_PORTABLE_INIT uses xor64 from line 1267 - it gets two uint8_t pointers